pulsar-broker/src/test/java/org/apache/pulsar/broker/service/OneWayReplicatorTest.java

```diff
@@ -652,4 +652,63 @@ public void testUnFenceTopicToReuse() throws Exception {
             admin2.topics().delete(topicName);
         });
     }
+
+    @Test
+    public void testNamespaceLevelReplicationRemoteConflictTopicExist() throws Exception {
+        final String topicName = BrokerTestUtil.newUniqueName("persistent://" + replicatedNamespace + "/tp");
+        // Verify: will get a not found error when calling "getPartitionedTopicMetadata" on a topic not exists.
+        try {
+            admin1.topics().getPartitionedTopicMetadata(topicName);
+            fail("Expected a not found error");
+        } catch (Exception ex) {
+            Throwable unWrapEx = FutureUtil.unwrapCompletionException(ex);
+            assertTrue(unWrapEx.getMessage().contains("not found"));
+        }
+        // Verify: will get a conflict error when there is a topic with different partitions on the remote side.
+        admin2.topics().createPartitionedTopic(topicName, 1);
+        try {
+            admin1.topics().createPartitionedTopic(topicName, 2);
```

Comment:
@Demogorgon314 

Separate a channel to discuss this, to make reading the context easier.

> I think we intentionally overlooked the result of creating a topic for the remote cluster.
> This PR may break the user's behavior of creating topic.
> According to your description, when using the default broker configuration, and the geo-replication is enabled on the namespace level, the remote cluster will create two non-partitioned topics, tenant/namespace/topic-partition-1 and tenant/namespace/topic-partition-2 by the geo producer. Is it right?
> If right, I would suggest adding a topic check before starting the replicator to make sure that the topic is the same between local and remote clusters, if they are the same, start the replicator, otherwise throw a log. You can also add this check in the replicator task.

#### Answer

> According to your description, when using the default broker configuration, and the geo-replication is enabled on the namespace level, the remote cluster will create two non-partitioned topics, tenant/namespace/topic-partition-1 and tenant/namespace/topic-partition-2 by the geo producer. Is it right?

No, regarding this test case, the messages that were sent to `partition-1` will never be copied to the remote cluster because `partition-1` can not be created successfully.

> I think we intentionally overlooked the result of creating a topic for the remote cluster.
> This PR may break the user's behavior of creating topic.

Without this fix, the broker eats the error, and users assume all things are fine. It is not correct.

Reply:
>According to your description, when using the default broker configuration, and the geo-replication is enabled on the namespace level, the remote cluster will create two non-partitioned topics, tenant/namespace/topic-partition-1 and tenant/namespace/topic-partition-2 by the geo producer. Is it right?

Sorry, I didn't consider your PR. 

If there isn't this PR,  the local cluster will do that.

> No, regarding this test case, the messages that were sent to `partition-1` will never be copied to the remote cluster because `partition-1` can not be created successfully.

There will still be some issues here:

1. If the remote cluster's topic equals the topic that will be created, how to handle it?

2. How to roll back a remote cluster's topic if the remote cluster's topic created the cluster successfully, but the local cluster's creation failed.

> Without this fix, the broker eats the error, and users assume all things are fine. It is not correct.

I understand here, so we have overlooked this error.

Reply:
> If the remote cluster's topic equals the topic that will be created, how to handle it?

Broker assumes created successfully when the topic is the same as trying to create. See https://github.com/apache/pulsar/pull/22577/files#diff-6902810fd24ff29ec34052bcc6a0de36d76107649148d945b80ec2be888e9042R683-R686

```java
if (topicMeta.partitions == numPartitions) {
    log.info("[{}] Skip created partitioned topic {} in cluster {},  because that {}",
            clientAppId(), topicName, cluster, unwrapEx2.getMessage());
    createRemoteTopicFuture.complete(null);
}
```

> How to roll back a remote cluster's topic if the remote cluster's topic created the cluster successfully, but the local cluster's creation failed.

Before creating topics at the remote cluster, the check for creating topics at the local cluster has been passed, so it is less probably to fail. Users should handle the error manually if created on the local cluster failed, for example, they can try again.

Reply:
I want to explain the replicator behavior.

The geo-replication is enabled on the namespace, `topic-1` has 1 partitions on the remote cluster, and you will create a `topic-1` topic with 3 partitions on the local cluster, which can be created.

When you send the message to `topic-1` from the local cluster, the replicator will be created, partition 1 and 2 topics will become non-partition topics on the remote cluster.

| local cluster | remote cluster |
| -- | -- |
| topic-1-parition-0(partitioned) | topic-1-partition-0(partitioned) |
| topic-1-parition-1(partitioned) | topic-1-parition-1(non-partition) |
| topic-1-parition-2(partitioned) | topic-1-parition-2(non-partition) |

Reply:
> When you send the message to topic-1 from the local cluster, the replicator will be created, partition 1 and 2 topics will become non-partition topics on the remote cluster.

No, the creation of partitions `1` and `2` will fail, and the replicator will get an error "Illegal topic partition name xxx with max allowed 1 partitions"